pages/controls/Certificate_and_Public_Key_Pinning.md

```diff
@@ -357,40 +357,142 @@ However, implementing pinning validation from scratch should be avoided,
+server certificate validation for **all** requests requiring validation from 
+the AppDomain. It is therefore advisable to check that the `sender` represents a request
+to the authority to which to apply certificate pinning, as the example above demonstrates.
+
+In .NET Framework 4.5 onwards, a validation callback can be set per HTTP request
+
+```
+// Encoded RSAPublicKey
+private static string PUB_KEY = "30818902818100C4A06B7B52F8D17DC1CCB47362" +
+	"C64AB799AAE19E245A7559E9CEEC7D8AA4DF07CB0B21FDFD763C63A313A668FE9D764E" +
+	"D913C51A676788DB62AF624F422C2F112C1316922AA5D37823CD9F43D1FC54513D14B2" +
+	"9E36991F08A042C42EAAEEE5FE8E2CB10167174A359CEBF6FACC2C9CA933AD403137EE" +
+	"2C3F4CBED9460129C72B0203010001";
 
 public static void Main(string[] args)
 {
-  ServicePointManager.ServerCertificateValidationCallback = PinPublicKey;
-  WebRequest wr = WebRequest.Create("https://encrypted.google.com/");
-  wr.GetResponse();
+	var wr = (HttpWebRequest)WebRequest.Create("https://encrypted.google.com/");
+    
+    // set server validation callback for this request only
+    wr.ServerCertificateValidationCallback = PinPublicKey;
+	wr.GetResponse();
 }
 
 public static bool PinPublicKey(object sender, X509Certificate certificate, X509Chain chain,
-                                SslPolicyErrors sslPolicyErrors)
+								SslPolicyErrors sslPolicyErrors)
 {
-  if (null == certificate)
-    return false;
+	// No errors
+	if (sslPolicyErrors == SslPolicyErrors.None)
+		return true;
+		
+	if (certificate == null)
+		return false;
+
+    var pk = certificate.GetPublicKeyString();
+	if (pk.Equals(PUB_KEY))
+		return true;
+
+	// There are SSL policy errors
+	return false;
+}
+```
+
+#### HttpClient
+
+When using `HttpClient`, a server validation callback can be set on
+`HttpClientHandler` that will affect **all** requests requiring
+validation made with the client instance.
 
-  String pk = certificate.GetPublicKeyString();
-  if (pk.Equals(PUB_KEY))
-    return true;
+```
+// Encoded RSAPublicKey
+private static string PUB_KEY = "30818902818100C4A06B7B52F8D17DC1CCB47362" +
+	"C64AB799AAE19E245A7559E9CEEC7D8AA4DF07CB0B21FDFD763C63A313A668FE9D764E" +
+	"D913C51A676788DB62AF624F422C2F112C1316922AA5D37823CD9F43D1FC54513D14B2" +
+	"9E36991F08A042C42EAAEEE5FE8E2CB10167174A359CEBF6FACC2C9CA933AD403137EE" +
+	"2C3F4CBED9460129C72B0203010001";
 
-  // Bad dog
-  return false;
+public static void Main(string[] args)
+{
+	MainAsync().Wait();
+}
+
+private static async Task MainAsync()
+{
+	var handler = new HttpClientHandler
+	{
+		ServerCertificateCustomValidationCallback = PinPublicKey
+	};
+
+	var client = new HttpClient(handler);
+	await client.GetAsync("https://encrypted.google.com/");
+}
+
+public static bool PinPublicKey(HttpRequestMessage requestMessage, X509Certificate2 certificate, X509Chain chain,
+								SslPolicyErrors sslPolicyErrors)
+{
+	// No errors
+	if (sslPolicyErrors == SslPolicyErrors.None)
+		return true;
```

Comment:
Another spot.

Reply:
Similar to above, the PinPublicKey method will be called for every HTTP request from the HttpClient instance, so for HTTPS requests to endpoints that present a valid certificate, validation should succeed.